### ISSUE TITLE
Bad file type

### ISSUE THREAD

**Issue by Michenux** [mafintosh/tar-stream#43]
Hi,

With the following archive :
http://download.oracle.com/otn-pub/java/jdk/7u79-b15/server-jre-7u79-windows-x64.tar.gz
The file type of the first entry is wrong. It should a directory, not a file :
jdk1.7.0_79/ file
jdk1.7.0_79/COPYRIGHT file
jdk1.7.0_79/LICENSE file
jdk1.7.0_79/README.html file
jdk1.7.0_79/release file

Here's my small program :
```javascript
var tar = require('tar-stream');
var fs = require('fs');

var extract = tar.extract();
extract.on('entry', function(header, stream, callback) {
  console.log(header.name + ' ' + header.type);
  stream.on('end', function() {
    callback() // ready for next entry
  })
  stream.resume() // just auto drain the stream
});

extract.on('finish', function() {
  // all entries read
  console.log('finish extract');
});

var readStream = fs.createReadStream('jdk.tar');
readStream.on('error', function(err) {
        console.log('read error', err);
});

readStream.pipe(extract);
```

**Comment by mafintosh** [comment]
The download link seems to have expired. Could you share a new one / tell how to download the tar?

**Comment by Michenux** [comment]
Sorry, direct link does not work.
Go on web page : http://www.oracle.com/technetwork/java/javase/downloads/server-jre7-downloads-1931105.html
Accept the licence
And download file server-jre-7u79-windows-x64.tar.gz

**Comment by mafintosh** [comment]
Thanks. I can reproduce. Will have a fix out soon

**Comment by mafintosh** [comment]
Should be fixed in 1.2.2. Thanks for reporting this

**Comment by Michenux** [comment]
Thanks a lot :)
